app/gui2/src/components/ToggleIcon.vue

```diff
@@ -0,0 +1,20 @@
+<script setup lang="ts">
+import SvgIcon from '@/components/SvgIcon.vue'
+import { ref } from 'vue'
+
+const props = defineProps<{ icon_name: String; modelValue?: Boolean }>()
+const emit = defineEmits<{
+  (e: 'update:modelValue', toggled_on: Boolean): void
+}>()
+
+const toggled_on = ref(props.modelValue ?? false)
```

Comment:
`camelCase`. Also I don't think that this internal state is necessary. Instead, I would make `modelValue` prop required and use it directly. `toggle` would then just emit. In normal operation, you expect that the model is updated in reaction to it (either by catching an event in parent component or automatically through `v-model`). When it is not handled, it's fine to let the component effectively do nothing on clicks. At least there is no confusion whether or not it was handled. Also it may just be the desired behavior for locked/disabled states.

Reply:
But in that case I would need to assign "toggled_on" class by hand, and with ref I can do it neatly.

Reply:
> But in that case I would need to assign "toggled_on" class by hand, and with ref I can do it neatly.

Applied @somebody1234 suggestion below, thank you!